Helper/TokenHelper.php

```diff
@@ -344,6 +346,21 @@ private function addHelper($type)
 
                                 return $dash ? substr((string) $value, 0, $dash) : $value;
                             },
+                            'plus4' => function ($value) {
+                                $dash = strrpos((string) $value, '-');
+                                if (!$dash) {
+                                    // No dash found, thus this is not a ZIP+4
+                                    return '';
+                                }
+                                $start = 1 - (strlen($value) - $dash);
+                                $value = substr((string) $value, $start);
+                                if (4 !== strlen($value)) {
+                                    // Not precisely 4 digits, thus this is not a ZIP+4
+                                    return '';
+                                }
+
+                                return $value;
+                            },
```

Comment:
Consider using regex?
Maybe something like: 
```
preg_match("/\d-([\d]{4})/", $value, $matches);
if(empty($matches) {
    return '';
}

return $matches[1];
```

Reply:
Much better... was in too much of a hurry.